src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -82,39 +84,80 @@ export default class HexagonLayer extends Layer {
     super(props);
   }
 
+  getShaders() {
+    return this.props.fp64 ? {
+      vs: readFileSync(join(__dirname, './hexagon-layer-64-vertex.glsl'), 'utf8'),
+      fs: readFileSync(join(__dirname, './hexagon-layer-fragment.glsl'), 'utf8'),
+      modules: ['fp64', 'project64', 'lighting']
+    } : {
+      vs: readFileSync(join(__dirname, './hexagon-layer-vertex.glsl'), 'utf8'),
+      fs: readFileSync(join(__dirname, './hexagon-layer-fragment.glsl'), 'utf8'),
+      modules: ['lighting']
+    };
+  }
+
   /**
    * DeckGL calls initializeState when GL context is available
    * Essentially a deferred constructor
    */
   initializeState() {
     const {gl} = this.context;
-    this.setState({model: this.getModel(gl)});
+    this.setState({model: this._getModel(gl)});
 
     const {attributeManager} = this.state;
     /* eslint-disable max-len */
     attributeManager.addInstanced({
       instancePositions: {size: 3, accessor: ['getCentroid', 'getElevation'], update: this.calculateInstancePositions},
       instanceColors: {size: 4, type: gl.UNSIGNED_BYTE, accessor: 'getColor', update: this.calculateInstanceColors}
     });
+
+    if (this.props.fp64) {
```

Comment:
We shouldn't need to do this check twice. It should be enough to do it in `updateState`, since updateState is called on first initialization as well (after `initializeState`). If that doesn't work, we might have a bug on our hands.